code/game/area/areas.dm

```diff
@@ -239,37 +239,33 @@ GLOBAL_LIST_EMPTY(teleportlocs)
  * Sends to all ai players, alert consoles, drones and alarm monitor programs in the world
  */
 /area/proc/poweralert(state, obj/source)
-	if (area_flags & NO_ALERTS)
+	if (area_flags & NO_ALERTS || state == poweralm || !istype(source))
 		return
-	if (state != poweralm)
-		poweralm = state
-		if(istype(source)) //Only report power alarms on the z-level where the source is located.
-			for (var/item in GLOB.silicon_mobs)
-				var/mob/living/silicon/aiPlayer = item
-				if (!state)
-					aiPlayer.cancelAlarm("Power", src, source)
-				else
-					aiPlayer.triggerAlarm("Power", src, cameras, source)
 
-			for (var/item in GLOB.alert_consoles)
-				var/obj/machinery/computer/station_alert/a = item
-				if(!state)
-					a.cancelAlarm("Power", src, source)
-				else
-					a.triggerAlarm("Power", src, cameras, source)
+	poweralm = state
+	for (var/mob/living/silicon/aiPlayer as anything in GLOB.silicon_mobs)
+		if (!state)
+			aiPlayer.cancelAlarm("Power", src, source)
+		else
+			aiPlayer.triggerAlarm("Power", src, cameras, source)
```

Comment:
```suggestion
	for (var/mob/living/silicon/ai_player as anything in GLOB.silicon_mobs)
		if (!state)
			ai_player.cancelAlarm("Power", src, source)
		else
			ai_player.triggerAlarm("Power", src, cameras, source)
```
We really could use diminishing the various random case variants across the codebase.